### PR TITLE
Added scripts of generating SDK references

### PR DESCRIPTION
**Pull request by chrisli30**
Adde this script to help us generate the SDK references from the `src/index.ts` file.

The generated documentation is copied to https://avaprotocol.org/docs/ethereum/develop/js-sdk

### CODE REVIEW

File: scripts/generate-docs.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+ */
+
+const fs = require("fs");
+
+const OUTPUT_DIR = "./scripts"; // The directory to output the generated documentation; setting is to ./scripts for now.
+const OUTPUT_FILENAME = "SDK_References.md";
+
+const methods = [
+  {
+    name: "authWithAPIKey",
```
Comment (v9n) [inline scripts/generate-docs.js:16]: I would not publicly document `authWithAPIKey` for now. Considering it's a private api for now. There is no way for an end-user to acquire this key. Once we implement an oauth process where application will register to get app id, and api key then we can start annoucen this. right now, no one can actually use this or even shoulduse this.

File: scripts/generate-docs.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
+/**
+ * This script generates the SDK reference documentation based on the methods array defined at the beginning of the file.
+ * For a generated example, please refer to https://avaprotocol.org/docs/ethereum/develop/js-sdk
```
Comment (v9n) [inline scripts/generate-docs.js:3]: hmm dead link

**Review by v9n — COMMENTED** [review]
looks good to me. 

is this a simplify jsdoc ?

i also like the approach of  https://docs.ethers.org/v6/api/  where they had getting started to guide end-user through a how-to style, and the api docs look up when you know how to use it, you just need to look up api doc.

i think eventually we should incoporate that kind of style into our https://avaprotocol.org/docs/ethereum/develop/api-reference

**Review by v9n — DISMISSED** [review]

File: scripts/generate-docs.js (hunk 2 above)
Comment (chrisli30) [reply scripts/generate-docs.js:3]: The link was on staging website.
https://oak-website-git-staging-avaprotocol.vercel.app/docs/ethereum/develop/js-sdk

I’m deploying it to main

File: scripts/generate-docs.js (hunk 1 above)
Comment (chrisli30) [reply scripts/generate-docs.js]: Right, that’s a good call. I removed it from this PR.

**Comment by chrisli30** [comment]
> looks good to me.
> 
> is this a simplify jsdoc ?
> 
> i also like the approach of https://docs.ethers.org/v6/api/ where they had getting started to guide end-user through a how-to style, and the api docs look up when you know how to use it, you just need to look up api doc.
> 
> i think eventually we should incoporate that kind of style into our https://avaprotocol.org/docs/ethereum/develop/api-reference

I agree. Created #33 to track this issue and add a Get Started section.

**Comment by chrisli30** [comment]
The updated doc is deployed and live on: https://avaprotocol.org/docs/ethereum/develop/js-sdk#usage-authentication-api-key-authentication

**Pull request merged**